zerver/tests/fixtures/slack_message_conversion.json

```diff
@@ -94,6 +94,11 @@
       "name": "format_emoji_test",
       "input": "*1️⃣ bold* some _1️⃣ italic_ word ~1️⃣ strike~",
       "conversion_output": "**1️⃣ bold** some *1️⃣ italic* word ~~1️⃣ strike~~"
+    },
+    {
+      "name": "overlapping_capture_group_test",
+      "input": "*abc*\n*def*\n*ghi*\n*jkl*\n\n*ABC ABC*",
+      "conversion_output": "**abc**\n**def**\n**ghi**\n**jkl**\n\n**ABC ABC**"
```

Comment:
If this is the only overlapping case we care about, using `re.MULTILINE` instead of look-ahead is sufficient, since then we're still matching against `^` and `$` which are zero-width.  Are there examples _without_ `\n` being involved, which we could use?

Reply:
yes, I believe strings like this would produce the same scenario: `**Hello**!**How are you**?`

Reply:
hmm, I tried running this with the existing regex and `re.MULTILINE` as another flag and it still failed:
```bash
======================================================================
FAIL: test_message_conversion_fixtures (zerver.tests.test_slack_message_conversion.SlackMessageConversion) (slack_message_conversion='overlapping_capture_group_test')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/srv/zulip/zerver/tests/test_slack_message_conversion.py", line 53, in test_message_conversion_fixtures
    self.assertEqual(converted_text, test["conversion_output"])
  File "/srv/zulip/zerver/tests/test_slack_message_conversion.py", line 21, in assertEqual
    raise AssertionError(
AssertionError: Actual and expected outputs do not match; showing diff.
- *abc*
+ **abc**
- *def*
+ **def**
- *ghi*
+ **ghi**
- *jkl*
+ **jkl**
  
- *ABC ABC*
+ **ABC ABC**


----------------------------------------------------------------------
Ran 3 tests in 0.028s

```

```python
def convert_link_format(text: str) -> tuple[str, bool]:
    """
    1. Converts '<https://foo.com>' to 'https://foo.com'
    2. Converts '<https://foo.com|foo>' to '[foo](https://foo.com)'
    """
    has_link = False
    for match in re.finditer(LINK_REGEX, text, re.VERBOSE | re.MULTILINE):
        slack_url = match.group(0)
        url_parts = slack_url[1:-1].split("|", maxsplit=1)
        # Check if there's a pipe with text after it
        if len(url_parts) == 2:
            converted_url = f"[{url_parts[1]}]({url_parts[0]})"
        else:
            converted_url = url_parts[0]

        has_link = True
        text = text.replace(slack_url, converted_url)
    return text, has_link
```
it'd still fail using the new regex capture groups without the look-ahead matching...

Reply:
(Deleted a comment because it was incorrect.)